zest/releaser/utils.py

```diff
@@ -494,6 +518,47 @@ def system(command, input=''):
     return result
 
 
+def retry_command(command):
+    """Run the command and possibly retry it.
+
+    Print interesting lines.
+
+    When we discover errors, we give three options:
+    - Abort
+    - Retry
+    - Continue
+
+    There is an error is there is a red color in the output.
+
+    It might be a warning, but we cannot detect the distinction.
+    """
+    result = system(command)
+    if Fore.RED not in result:
+        show_interesting_lines(result)
+        return result
+    # There are warnings or errors. Print the complete result.
+    print(result)
+    question = ("There were errors or warnings. Are you sure "
+                "you want to continue? [Yes/No/Retry]")
+    if AUTO_RESPONSE:
+        msg = ("The question '%s' requires a manual answer, but "
+               "we're running in --no-input mode.")
+        msg = msg % question
+        raise RuntimeError(msg)
+    while True:
+        input = get_input(question)
+        if input:
+            input = input.lower()
+            if input == 'y':
+                # Accept the error, continue with the program.
+                return result
+            if input == 'n':
+                sys.exit(1)
```

Comment:
We are indeed using this only in PyPI upload.

I'm not sure there is anything useful we can tell people here if they choose not to continue, except a too obvious 'please investigate'.

We could retry automatically and that could sometimes help. So we could put that in. But the problem could also be wrong PyPI credentials, in which case a retry will not help. I now release some packages as user 'maurits' and some as user 'plone'. If the command fails, I can now edit my `~/.pypirc` and retry and it will actually work.

With this question there is no default answer. So if you just hit enter, you will simply see the same question again.

Would it help to change the possible answers? For example: Retry / Continue / Quit?